### PR TITLE
WIP: Optimize config applying

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/33335

### CODE REVIEW

File: tests/integration/pilot/common/traffic.go [hunk 1]
```diff
@@ -193,6 +241,98 @@ func (c TrafficTestCase) RunForApps(t framework.TestContext, apps echo.Instances
...
+	v, f := p["_dst_written"]
+	return f && v != nil && v.(bool)
+}
+
+func (p tmplParams) WroteSrc() bool {
+	v, f := p["_src_written"]
+	return f && v != nil && v.(bool)
+}
+
+func findReferences(t test.Failer, n int, vars func(src echo.Callers, dest echo.Instances) map[string]interface{}, config string, echoT *echotest.T) (bool, bool) {
```
Comment (stevenctl) [inline tests/integration/pilot/common/traffic.go:326]: cool

Comment (stevenctl) [reply tests/integration/pilot/common/traffic.go:326]: I assume this works for nested access?

**Comment by stevenctl** [comment]
Not sure if your run was lucky – looks like postsubmit is anywhere from 18m to ~25m, but seems like a really good improvement. 

Maybe a future cleanup would be moving some of our traffic test logic into echotest since we essentially have 2 frameworks here.

**Comment by howardjohn** [comment]
> Maybe a future cleanup would be moving some of our traffic test logic into echotest since we essentially have 2 frameworks here.


Strongly agree

**Pull request closed without merging**